aggsender/aggsender.go

```diff
@@ -825,3 +858,26 @@ func NewCertificateInfoFromAgglayerCertHeader(c *agglayer.CertificateHeader) *ty
 	}
 	return res
 }
+
+// getLastSentBlockAndRetryCount returns the last sent block of the last sent certificate
+// if there is no previosly sent certificate, it returns 0 and 0
+func getLastSentBlockAndRetryCount(lastSentCertificateInfo *types.CertificateInfo) (uint64, int) {
+	if lastSentCertificateInfo == nil {
+		return 0, 0
+	}
+
+	retryCount := 0
+	previousToBlock := lastSentCertificateInfo.ToBlock
+
+	if lastSentCertificateInfo.Status == agglayer.InError {
+		// if the last certificate was in error, we need to resend it
+		// from the block before the error
+		if lastSentCertificateInfo.FromBlock > 0 {
+			previousToBlock = lastSentCertificateInfo.FromBlock - 1
+		}
+
+		retryCount = lastSentCertificateInfo.RetryCount + 1
+	}
+
+	return previousToBlock, retryCount
+}
```

Comment:
Rename to `lastSentBlock`?
```suggestion
	lastSentBlock := lastSentCertificateInfo.ToBlock

	if lastSentCertificateInfo.Status == agglayer.InError {
		// if the last certificate was in error, we need to resend it
		// from the block before the error
		if lastSentCertificateInfo.FromBlock > 0 {
			lastSentBlock = lastSentCertificateInfo.FromBlock - 1
		}

		retryCount = lastSentCertificateInfo.RetryCount + 1
	}

	return lastSentBlock, retryCount
```